file_handler_test.cpp

```diff
@@ -9,43 +9,56 @@
 namespace http {
 namespace server {
 
+
 TEST(SimpleRequestTest, Docroot) {
-	NginxConfig config;
-	Request req;
+	
 	Response resp;
-	//std::string sample_request="GET /test.txt HTTP/1.1 \nHost: localhost:12345";
-	std::string sample_request="GET /test.txt HTTP/1.1\nHost:localhost";
-	req.Parse(sample_request);
+	//Setting up nested part of config block that specifies mapping
+	NginxConfig* parsedConfig = new NginxConfig();
+	NginxConfigStatement *st = new NginxConfigStatement();
+	st->tokens_.push_back("root");
+	st->tokens_.push_back("files");
+	auto pointer = std::shared_ptr<NginxConfigStatement>(st);
+	parsedConfig->statements_.push_back(pointer);
+
+	//Parsing Request
+	std::unique_ptr<Request> req = Request::Parse("GET /static1/test.txt HTTP/1.1\r\n\r\n");
 	file_handler *f= new file_handler();
-	f->Init("", config);
-	f->HandleRequest(req, &resp);
-	std::string test = "Content-Length: " + std::to_string(test.length()) + "\r\n";
-	test += "Content-Type: text/plain\r\n";
-	test += "Just a test file to check webserver response in the Integration Test.\n";
-	EXPECT_EQ(req.method(), "GET");
-	EXPECT_EQ(req.uri(), "/test.txt");
+	f->Init("/static1", *parsedConfig);
+	std::cout << parsedConfig->statements_.size() <<std::endl;
+	Request r = *req;
+	f->HandleRequest(r, &resp);
+	std::string test = "HTTP/1.1 200 OK\r\nContent-Length: 70\r\nContent-Type: text/plain\r\nJust a test file to check webserver response in the Integration Test.\n";
 	EXPECT_EQ(resp.ToString(), test);
 }
 
 TEST(BadRequestTest, Docroot) {
-	NginxConfig config;
-	Request req;
+	NginxConfig* parsedConfig = new NginxConfig();
+	NginxConfigStatement *st = new NginxConfigStatement();
+	st->tokens_.push_back("root");
+	st->tokens_.push_back("files");
+	auto pointer = std::shared_ptr<NginxConfigStatement>(st);
+	parsedConfig->statements_.push_back(pointer);
 	Response resp;
-	req.Parse("GET abcd.txt HTTP/1.1 \nHost: localhost:12345");
+	std::unique_ptr<Request> req = Request::Parse("GET test.txt HTTP/1.1\r\n\r\n");
```

Comment:
For this test, how can we tell the difference between bad request and 404 not found. Is it because the path "test.txt" is not specified as "/test.txt"? Or is it something else entirely?